packages/kbn-grouping/src/components/accordion_panel/group_stats.tsx

```diff
@@ -83,45 +91,50 @@ const GroupStatsComponent = <T,>({
             </span>
           </EuiFlexItem>
         );
-      }),
+      }) ?? [],
     [statRenderers]
   );
 
   const takeActionMenu = useMemo(
-    () => (
-      <EuiFlexItem grow={false}>
-        <EuiPopover
-          anchorPosition="downLeft"
-          button={
-            <EuiButtonEmpty
-              data-test-subj="take-action-button"
-              onClick={onButtonClick}
-              iconType="arrowDown"
-              iconSide="right"
-            >
-              {TAKE_ACTION}
-            </EuiButtonEmpty>
-          }
-          closePopover={() => setPopover(false)}
-          isOpen={isPopoverOpen}
-          panelPaddingSize="none"
-        >
-          <EuiContextMenuPanel items={takeActionItems} />
-        </EuiPopover>
-      </EuiFlexItem>
-    ),
+    () =>
+      takeActionItems.length ? (
+        <EuiFlexItem grow={false}>
+          <EuiPopover
+            anchorPosition="downLeft"
+            button={
+              <EuiButtonEmpty
+                data-test-subj="take-action-button"
+                onClick={onButtonClick}
+                iconType="arrowDown"
+                iconSide="right"
+              >
+                {TAKE_ACTION}
+              </EuiButtonEmpty>
+            }
+            closePopover={() => setPopover(false)}
+            isOpen={isPopoverOpen}
+            panelPaddingSize="none"
+          >
+            <EuiContextMenuPanel items={takeActionItems} />
+          </EuiPopover>
+        </EuiFlexItem>
+      ) : null,
     [isPopoverOpen, onButtonClick, takeActionItems]
   );
 
   return (
     <EuiFlexGroup
       data-test-subj="group-stats"
       key={`stats-${bucketKey}`}
-      gutterSize="none"
+      gutterSize="m"
       alignItems="center"
     >
-      {statsComponent}
-      {takeActionMenu}
+      {[...statsComponents, takeActionMenu].filter(Boolean).map((component, index, { length }) => (
```

Comment:
This allows us to hide the `Take actions` menu (as requested by O11y) without showing the last separator

Reply:
Nice! Is this new logic tested?

Reply:
Added test cases in dff4e12890ea2e13861d92e1faa72c6daa7b84d3